metecho/api/gh.py

```diff
@@ -79,6 +80,20 @@ def get_repo_info(user, repo_id=None, repo_owner=None, repo_name=None):
     return gh.repository_with_id(repo_id)
 
 
+def get_cached_user(gh: GitHub, username: str) -> users.User:
+    """
+    Get a GitHub user by username. Results are cached to stay under API limits.
+    """
+    key = f"gh_user_{username}"
+    user_dict = cache.get(key)
+    if user_dict is not None:
+        return users.User.from_dict(user_dict, gh.session)
+
+    user = gh.user(username)
+    cache.set(key, user.as_dict(), timeout=60 * 60 * 24)  # 1 day
+    return user
```

Comment:
@jgerigmeyer we're caching users locally so we don't hit the GitHub API if not needed. This should make the related job execute faster after the first run and avoid querying the API for users that collaborate on multiple projects. I haven't tested if it's actually necessary to stay under API limits, but I figured it can't hurt either.

Reply:
Yeah, I think that's a good start. @davisagli Do you have concerns about this? (Context: To get full names of GH users, we need to do individual requests to the GH API for each user, in addition to getting the one big list of all collaborators.)

Reply:
This is a cache backed by redis, as currently configured, right? I think it's all right. Good to have a TTL on the cache entry and 1 day sounds fine.

I think it would also be okay to only include a full name for users who have previously logged into Metecho (in which case we could get it as part of the oauth login and store it on the User)

Reply:
That would remove the need to expand all collaborators as we do now. In the context of this story and mockup, does it make sense to only display names for some users?

Reply:
@davisagli and yes, this is the cache backed by redis

Reply:
@jerivas It's certainly a better UX if we can show the name for every user in the list, but if we end up hitting rate limits I think it's okay to skip that part for now.

Reply:
I think ideally we'd have names for all GH users in the list, so let's start with this and go the other route only if we run into issues.